app/Http/Controllers/Supplier/ProductController.php

```diff
@@ -4,34 +4,47 @@
 
 use App\Http\Requests\EditProductFormRequest;
 use App\Http\Requests\ProductFormRequest;
-use App\Models\Category;
-use App\Models\Image;
-use App\Models\Product;
 use App\Http\Controllers\Controller;
+use App\Repositories\Category\CategoryRepositoryInterface;
+use App\Repositories\Image\ImageRepositoryInterface;
+use App\Repositories\Product\ProductRepositoryInterface;
 use Illuminate\Support\Facades\Auth;
 use Illuminate\Support\Facades\DB;
 use Exception;
 use RealRashid\SweetAlert\Facades\Alert;
 
 class ProductController extends Controller
 {
+    protected $productRepo;
+    protected $categoryRepo;
+    protected $imageRepo;
+
+    public function __construct(ProductRepositoryInterface $productRepo,
+        CategoryRepositoryInterface $categoryRepo,
+        ImageRepositoryInterface $imageRepo)
+    {
```

Comment:
```suggestion
        ImageRepositoryInterface $imageRepo
    ) {
```
(luom) vừa cmt pull trc xong